tensorflow/core/framework/dataset.cc

```diff
@@ -496,6 +496,42 @@ Status DatasetBaseIterator::GetNext(IteratorContext* ctx,
   return s;
 }
 
+Status DatasetBaseIterator::Skip(IteratorContext* ctx, int num_to_skip,
+                                 bool* end_of_sequence, int* num_skipped) {
+  profiler::TraceMe activity([&] { return BuildTraceMeName(); },
+                             profiler::TraceMeLevel::kInfo);
+  DVLOG(3) << prefix() << " Skip enter";
+  RecordStart(ctx, /*stop_output=*/true);
+  Status s = SkipInternal(ctx, num_to_skip, end_of_sequence, num_skipped);
+  if (s.ok() && !*end_of_sequence) RecordElement(ctx);
```

Comment:
After skipping `num_skipped` elements, this will record the iterator as having produced one element. This could confuse the autotuning implementation, which uses these recorded times to decide how to tune parallelism levels and buffer sizes. We need to figure out a way to make autotuning interact well with skipping. Looping in @jsimsa, who may have a recommendation for what to do here.

Reply:
Sorry for the long turnaround on this PR @zhuzilin. I was out of office for extended period of time and @aaudiber was waiting for my feedback.

The autotuning implementation assumes that the sum of CPU time spent executing logic of a given iterator `GetNext(Internal)` call across all of its invocations divided by the number of computed elements (as recorded by `RecordElement`) represents the average time to process a single element. My concern is that the current PR creates an opportunity for this assumption to be violated. 

It seems to me that the element should be recorded only if the `GetNextInternal`-based default implementation of `SkipInternal` is invoked. tf.data kernels which provide "efficient" version of `SkipInternal` should not increment the element count (which means we will assume that skipping an element has negligible cost compared to computing it).

So my suggestion would be to move `RecordElement` to the default implementation of `SkipInternal` and explain in a comment there why we do that.

Reply:
I've moved the `RecordElement` and added the comment.